server/src/test/java/io/crate/integrationtests/GeoShapeIntegrationTest.java

```diff
@@ -63,14 +63,20 @@ public void prepare() throws Exception {
                 "  shape geo_shape," +
                 "  shapes array(geo_shape)" +
                 ") with (number_of_replicas=0)");
+        execute("CREATE TABLE bkd_shaped (" +
+                "  id long primary key," +
+                "  shape geo_shape INDEX USING bkdtree" +
+                ") with (number_of_replicas=0)");
```

Comment:
Can we add a new field, indexed with bkdtree, into existing table instead of creating a new table?

Reply:
I'm now thinking whether it's a good idea to allow mixed indexing in a single table. Yes, as mentioned above, MatchPredicate  can be done BWC but not sure how build Lucene query in `WithinFunction.toQuery `for smth like `within(bkd_shape, geohash_shape)`. Even is possible, feels like error prone.

I think we should completely restrict non-BKD tree, mention that in breaking changes (that only BKD in supported from now on), add analyzer unit tests to verify that geohash/quadtree now throws an error. 

Then we don't need new table `bkd_shaped` and rather need to adjust all tests which are currently using geohash/quadtree to use BKD and they should be still passing.

However, we still need old indexing code in the `GeoShapeIndexer`/`GeoFieldMapper/GeoReference` for BWC reasons. We want to support rolling upgrades or insert into old tables, snapshotted in the old cluster and restored into new cluster). 

Maybe in 6.0 we can completely remove old code and say that we don't allow restore 5.* snapshots into 6.* clusters.

@seut, @mfussenegger, @matriv  wdyt?

Reply:
UPD:
`within(shape1, shape2)` falls to `GenericFunctionQuery` -> full scan anyway (see https://github.com/crate/crate/pull/14942#discussion_r1373106209)  but still believe we shouldn't mix old/new indexing.

Reply:
replied in https://github.com/crate/crate/pull/14942#issuecomment-1798656694